Automattic/wp-calypso
# Show the expert banner across the theme tabs except for My Themes

#### Proposed Changes

This PR exposes the expert banner across the theme tabs except for My Themes per request: p2-pau2Xa-4nq, i.e. this banner:
<img width="949" alt="image" src="https://user-images.githubusercontent.com/1842898/191924209-e86da4a1-9ad6-4b44-8213-060c29648d96.png">

Currently, it's only shown in All Themes tab of `/themes`, along with another plan upsell banner:
<img width="953" alt="image" src="https://user-images.githubusercontent.com/1842898/191924388-ab2b6bee-9999-433e-9c7a-ff942fa5ac12.png">

In this PR, the banner will be shown in the Recommended tab, the Trending tab, the All Themes tab. The plan upsell banner will show after dismissing the expert banner to avoid showing two banners at the same time: p2-pau2Xa-4nq#comment-12477

Some more details:

* Add a selector of `<DismissibleCard/>` so it's possible to check for its dismissing state. Some refactor is done along with it.
* The utm_campaign of the expert banner in the Recommmended and Trending tabs is `theme-rec-tre`, while it is `theme-all` in the All Themes tab. For all the tabs, `utm_source` is `WordPresscom` and `utm_medium` is `automattic_referred`.
* The `location` value for each tab has been updated so we can do granular analysis in Tracks if we want.
* The hard-coded strings are replaced by the constants defined in `this.tabFilters`.

#### Testing Instructions

1. Log in as a user who has never dismissed the expert banner. The plan-tier it owns has to be lower than the Business plan  so the plan upsell banner is eligible to show.
2. Go to `/theme`.
3. In the Recommended tab, the expert banner is shown, and the utm parameters are set as described above.
4. Dismissing the expert banner, the plan upsell banner is shown.
5. Refreshing, it should still be the pan upsell banner.
6. Log in as a user who has installed a custom theme. The fastest way of doing so might be to create an eCommerce site.
7. Going to `/theme`, the My Themes tab shouldn't have the expert banner.

#### Pre-merge Checklist
